File: Math & Game Theory & Simulation/Sum of Imbalance Numbers of All Subarrays/solution_O(n^2).py
class Solution:
    def sumImbalanceNumbers(self, nums: List[int]) -> int:
        n = len(nums)
        result = 0
        for i in range(n):
            s = set([nums[i]])
            cur = 0
            for num in nums[i+1:]:
                if num in s:
                    pass
                elif num-1 in s and num+1 in s:
                    cur -= 1
                elif num-1 not in s and num+1 not in s:
                    cur += 1
                s.add(num)
                result += cur
        return result
                    
                

            
                        

File: Array/Count Number of Teams/solution_O(n^2).cpp
class Solution {
public:
    int numTeams(vector<int>& rating) {
        int n = rating.size();
        int result = 0;
        for(int j = 1; j < n-1; ++j) {
            int lt = 0, gt = 0;
            for(int i = 0; i < j; ++i) {
                lt += rating[i] < rating[j];
                gt += rating[i] > rating[j];
            }
            for(int k = j+1; k < n; ++k) {
                if(rating[k] > rating[j]) result += lt;
                if(rating[k] < rating[j]) result += gt;
            }
        }
        return result;
    }
};
